Fix render_math condition

martinRenou opened the pull request:
Copied by @CAM-Gerlach from [the comment below](https://github.com/spyder-ide/docrepr/pull/25#discussion_r762839498)

> `if sphinx.__version__ < "1.1" or not render_math` seems to fail later if `render_math` is False: The reason is that latest sphinx versions seem to have removed jsmath, though we try to load it when `render_math` is False.

Related to #26

martinRenou committed:
Fix render_math condition

`if sphinx.__version__ < "1.1" or not render_math` seems to fail later
if render_math is False: The reason is that latest sphinx seem to have
removed jsmath.

martinRenou commented on docrepr/sphinxify.py:
```diff
@@ -203,10 +203,11 @@ def generate_extensions(render_math):
     """Generate list of Sphinx extensions"""
     # We need jsmath to get pretty plain-text latex in docstrings
     extensions = []
-    if sphinx.__version__ < "1.1" or not render_math:
```
`if sphinx.__version__ < "1.1" or not render_math` seems to fail later if `render_math` is False: The reason is that latest sphinx versions seem to have removed jsmath, though we try to load it when `render_math` is False.

CAM-Gerlach replied:
Makes sense, thanks! Copied this to your empty OP.

CAM-Gerlach commented on docrepr/sphinxify.py:
```diff
@@ -203,10 +203,11 @@ def generate_extensions(render_math):
-    else:
-        extensions = ['sphinx.ext.mathjax']
+    if render_math:
+        if sphinx.__version__ < "1.1":
+            extensions = ['sphinx.ext.jsmath']
+        else:
+            extensions = ['sphinx.ext.mathjax']
```
```suggestion
        extensions = ['sphinx.ext.mathjax']
```

Sphinx 1.1 was released over a decade ago, and most likely `docrepr` doesn't even work with Sphinx that old anymore anyway, so we may as well just simplify this...

martinRenou replied:
Yep :) There are mentions to jsmath in the config that we should maybe remove as well?

CAM-Gerlach reviewed (commented):
Good catch, thanks! I just had one suggestion on the implementation.

martinRenou commented:
Thanks for the review :) I took your comment into account and removed any jsmath mention.

I also added a lower bound pin on sphinx since mathjax was not there prior to 1.1.

CAM-Gerlach commented on docrepr/sphinxify.py:
```diff
@@ -201,16 +201,12 @@ def init_template_vars(oinfo):
     # be rendered correctly (see Spyder Issue #1138)
-    extensions.append('sphinx.ext.autosummary')
+    extensions = ['sphinx.ext.autosummary']
+
```
```suggestion

     # We need mathjax to get pretty plain-text latex in docstrings
```

Seems like this comment from above still applies, with a tweak?

CAM-Gerlach reviewed (commented):
Thanks @martinRenou ! I just have one trivial suggestion for a comment that was removed, which I believe should still apply to mathjax (with a tweak), sorry; otherwise this looks ready, thanks!

martinRenou committed:
Add back comment for mathjax

Co-authored-by: CAM Gerlach <CAM.Gerlach@Gerlach.CAM>

martinRenou commented:
Thanks :)

CAM-Gerlach reviewed (approved):
LGTM now, thanks @martinRenou !
